pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -630,9 +634,10 @@ func (cloud *CloudProvider) ResizeDisk(ctx context.Context, volKey *meta.Key, re
 	sizeGb := cloudDisk.GetSizeGb()
 	requestGb := common.BytesToGb(requestBytes)
 
-	// If disk is already of size equal or greater than requested size, we simply return
+	// If disk is already of size equal or greater than requested size, we
+	// simply return the found size
 	if sizeGb >= requestGb {
-		return requestBytes, nil
+		return sizeGb, nil
```

Comment:
@gnufied also identified this as one of the problematic driver responses to resize. We should be returning the actual size, not the requested size.

Reply:
Nm it's too early in the morning to read properly :-)

Reply:
Can we add a unit test for this case that verifies we return actual size instead of request?

Reply:
it's really hard to unit test this package right now as there was no easy way to fake out the actual cloud provider layer. So this "is" the cloud provider layer that I hacked together that we use as an interface to fake out. Suggestions welcome. It's in the backlog to switch over to the autogenerated cloud provider client layer that we switched over to in Kubernetes a while back - that provided the ability to fake out calls and unit test these types of things effectively.